frontend/src/metabase/parameters/components/Parameters/syncQueryParamsWithURL.js

```diff
@@ -0,0 +1,101 @@
+import Dimension from "metabase-lib/lib/Dimension";
+
+export const syncQueryParamsWithURL = props => {
+  props.commitImmediately
```

Comment:
Using a `prop` that is initiated by `frontend/src/metabase/query_builder/components/__mocks__/NativeQueryEditor.jsx`.

As best as I can tell, it marks a clear distinction between internal and public questions — without having to drill down a new prop.